### DISCUSSION TITLE
Error: failed to decode Kubernetes YAML

### DISCUSSION THREAD

**Discussion by unalerdem** [fluxcd/flux2#2169] [Q&A]
The steps I followed:
(1) Create a Helm Chart and modify values.yaml accordingly.
(2) Create a Flux HelmRelease and add helm-release.yaml in chart folder.

`kustomization/staging.flux-system
failed to decode Kubernetes YAML from /tmp/staging1594489832/apps/my-service/Chart.yaml: missing kind in object {{v2 } {{ } map[] map[]}}`

Am I doing something wrong that flux cant decode Chart.yaml because Kind is missing?

**Comment by somtochiama** [comment]
If your chart is stored in a git repository, you have to
Create Flux `GitRepository` -> Create `HelmRelease` that references the git repository

Please see this docs on [Managing helm release](https://fluxcd.io/docs/guides/helmreleases/)

**Reply by unalerdem** [reply]
Thanks for your answer.

The error indicates that the flux can't parse the Chart.yaml, so i guess the flux knows where to look at for the release.

In the repo for flux-system, the GitRepository and Kustomization is declared for the staging repo where Helm Charts exist.

~~~
---
apiVersion: source.toolkit.fluxcd.io/v1beta1
kind: GitRepository
metadata:
  name: staging
  namespace: flux-system
spec:
  interval: 30s
  ref:
    branch: main
  url: <git-repo-url>

---
apiVersion: kustomize.toolkit.fluxcd.io/v1beta2
kind: Kustomization
metadata:
  name: staging
  namespace: flux-system
spec:
  interval: 10m0s
  path: ./
  prune: true
  sourceRef:
    kind: GitRepository
    name: staging
~~~

In the staging repo, I added my-service-helm-release.yaml created with the flux cli.

~~~
---
apiVersion: helm.toolkit.fluxcd.io/v2beta1
kind: HelmRelease
metadata:
  name: my-service
  namespace: flux-system
spec:
  chart:
    spec:
      chart: apps/my-service
      sourceRef:
        kind: GitRepository
        name: staging
  interval: 10m0s
  targetNamespace: staging
~~~

**Reply by hiddeco** [reply]
The problem is that while the kustomize-controller scans the root directory for YAML files, it thinks the charts located relatively to the repository root (as configured in the `Kustomization`) are Kubernetes resources as well, while they are in fact templates or metadata files.

You either need to ensure the `Kustomization` uses a relative path which does not contain charts, or fabricate a `kustomization.yaml` in the root which instructs what specifically should be included (to prevent the controller from trying to figure it out and tripping over the charts).

**Reply by unalerdem** [reply]
Thank you for directing me to the right point.

I fixed the issue by putting helm release files in a release folder and setting the path to that folder in Kustomization.yaml.

~~~
---
apiVersion: kustomize.toolkit.fluxcd.io/v1beta2
kind: Kustomization
metadata:
  name: staging
  namespace: flux-system
spec:
  interval: 10m0s
  path: ./releases <--- here
  prune: true
  sourceRef:
    kind: GitRepository
    name: staging
~~~
